pkg/apis/application/v1alpha1/types_test.go

```diff
@@ -2635,3 +2635,83 @@ func TestGetCAPath(t *testing.T) {
 		assert.Empty(t, path)
 	}
 }
+
+func TestAppProjectIsSourceNamespacePermitted(t *testing.T) {
```

Comment:
Would probably be worthwhile to add a glob pattern test.

Reply:
Agreed. Added.